include/bactria/Event.hpp

```diff
@@ -13,62 +13,180 @@
+             */
             Event(std::string name, std::uint32_t color = bactria::color::orange, Category category = Category{})
             : Marker(std::move(name), color, std::move(category))
             {}
 
+            /**
+             * \brief The copy constructor.
+             *
+             * The copy constructor will generate a new event with the same properties as the original.
+             *
+             * \param other The Event to copy the properties from.
+             */
             Event(const Event& other)
             : Marker(other)
-            , m_handle{plugin::create_event(m_name.c_str(), m_color, m_category.id)}
+            , m_handle{detail::is_activated() ? plugin::create_event(m_color, m_category.get_c_name(), m_category.get_id()) : nullptr}
+            , m_action{other.m_action}
             {}
 
+            /**
+             * \brief The copy assignment operator.
+             *
+             * The copy assignment operator will generate a new event with the same properties as the original.
+             *
+             * \param rhs The Event to copy the properties from.
+             */
             auto operator=(const Event& rhs) -> Event&
             {
                 Marker::operator=(rhs);
-                m_handle = plugin::create_event(m_name.c_str(), m_color, m_category.id);
+
+                if(detail::is_activated())
+                    m_handle = plugin::create_event(m_color, m_category.get_c_name(), m_category.get_id());
+                else
+                    m_handle = nullptr;
+
+                m_action = rhs.m_action;
 
                 return *this;
             }
             
-            Event(Event&& other) noexcept = default;
-            auto operator=(Event&& rhs) noexcept -> Event& = default;
+            /**
+             * \brief The move constructor.
+             *
+             * The move constructor will move the contents of the original Event into this Event. The original
+             * Event will be in an invalid state; using it is undefined behaviour.
+             *
+             * \param other The Event to be moved.
+             */
+            Event(Event&& other) noexcept
+            : Marker(std::forward<Event>(other)), m_action{std::move(other.m_action)}
+            {
+                std::swap(m_handle, other.m_handle);
+            }
 
+            /**
+             * \brief The move assignment operator.
+             *
+             * The move assignment operator will move the contents of the original Event into this Event. The original
+             * Event will be in an invalid state; using it is undefined behaviour.
+             *
+             * \param rhs The Event to be moved.
+             */
+            auto operator=(Event&& rhs) noexcept -> Event&
+            {
+                Marker::operator=(std::forward<Event>(rhs));
+                std::swap(m_handle, rhs.m_handle);
+                m_action = std::move(rhs.m_action);
```

Comment:
The move might be ok here, but I would still do a `std::swap` just to follow the swap pattern for move implementation.